### PR TITLE
Kafka 5265

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

**Comment by ijuma** [comment]
retest this please

File: checkstyle/import-control.xml [hunk 1]
```diff
@@ -109,9 +118,11 @@
 
     <subpackage name="requests">
       <allow pkg="org.apache.kafka.clients.admin" />
```
Comment (ijuma) [inline checkstyle/import-control.xml:120]: Do we still need this?

File: clients/src/main/java/org/apache/kafka/clients/admin/CreateTopicsResults.java [hunk 2]
```diff
@@ -25,10 +25,10 @@
  * The result of newTopics.
  */
 @InterfaceStability.Unstable
-public class CreateTopicResults {
+public class CreateTopicsResults {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/admin/CreateTopicsResults.java]: Thanks for fixing this, I had it in one of my to-do lists somewhere. :) Also, while we are doing this, should it be `CreateTopicsResults` or `CreateTopicsResult`? I prefer the latter a little as it makes it easier to name lists of this class. But maybe the `results` method may look a little out of place (and we may need a new name for it).

**Review by ijuma — COMMENTED** [review]
Thanks for the PR, left a couple of questions. Double-checking the `TopicConfigs` now.

File: clients/src/main/java/org/apache/kafka/common/config/TopicConfigs.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+ * topic using the AdminClient.
+ *
+ * This is a public API, so we should not remove or alter keys without a discussion and a deprecation period.
+ * Eventually this should replace LogConfig.scala.
+ *
+ * Note that some of these are subtly different from the configuration keys that control
+ * the broker defaults.  For example, to set the default broker cleanup policy, you set log.cleanup.policy,
+ * but to set the per-topic cleanup policy, you set cleanup.policy.
+ */
+public class TopicConfigs {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/config/TopicConfigs.java:31]: Shouldn't this be `TopicConfig` to be consistent with `ConsumerConfig`, `ProducerConfig`, etc.?

File: checkstyle/import-control.xml (hunk 1 above)
Comment (cmccabe) [reply checkstyle/import-control.xml:120]: good call, removed.

File: clients/src/main/java/org/apache/kafka/clients/admin/CreateTopicsResults.java (hunk 2 above)
Comment (cmccabe) [reply clients/src/main/java/org/apache/kafka/clients/admin/CreateTopicsResults.java]: I'm pretty neutral on `CreateTopicsResults` vs `CreateTopicsResult.`  Creating a `List<CreateTopicsResults>` is consistent with some of the other classes we have, though, right?

Comment (cmccabe) [reply clients/src/main/java/org/apache/kafka/clients/admin/CreateTopicsResults.java]: OK, let's switch it to singluar

File: clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java [hunk 4]
```diff
@@ -197,7 +200,6 @@ public ApiVersionsResults apiVersions(Collection<Node> nodes) {
     public abstract ApiVersionsResults apiVersions(Collection<Node> nodes, ApiVersionsOptions options);
 
     /**
-<<<<<<< HEAD
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java]: Note to self: remove this conflict marker before merging.

**Comment by cmccabe** [comment]
Fixed merge conflict

**Comment by cmccabe** [comment]
retest this please

**Comment by ijuma** [comment]
LGTM, merged to trunk and 0.11.0.

**Pull request closed without merging**